### DISCUSSION TITLE
Headless CMS should be Database Agnostic

### DISCUSSION THREAD

**Discussion by firozemzr** [OSBDAlliance/headlesscms#6] [Ideas]
Headless CMS should be Database Agnostic. It should be able to connect to major relational and non-relational databases at the back end seamlessly. The Project should provide database scripts to create and upgrade database for different engines - postgresql, mysql, mongo, mssql, oracle etc.

**Comment by mohammad-arif-rahim** [comment]
if Database Agnostic means that it an application does not have to depend on the CMS to call database, then it is alright.  But anything else will have to be handled case by case. Usually everyone keeps the control of their database to themselves. It is also a security concern. There will be requirements of European Union's GDRP compliance signing for data security. It is not that it cannot be done, but it brings legal issues as well as technical issues as well as Backup and recovery.

**Reply by firozemzr** [reply]
Database agnostic means, user can decide about his own database engine, we should not tie it with any specific engine like mysql, mssqly, mongo etc..

Deployment should be users choice.
